diff --git a/package/kernel/linux/modules/other.mk b/package/kernel/linux/modules/other.mk
index 89f4ebc8..d714c6b3 100644
--- a/package/kernel/linux/modules/other.mk
+++ b/package/kernel/linux/modules/other.mk
@@ -975,12 +975,17 @@ $(eval $(call KernelPackage,ikconfig))
 define KernelPackage/zram
   SUBMENU:=$(OTHER_MENU)
   TITLE:=ZRAM
+  DEPENDS:=+LINUX_6_12:kmod-lib-lzo +LINUX_6_12:kmod-lib-lz4 \
+           +LINUX_6_12:kmod-lib-zstd
   KCONFIG:= \
 	CONFIG_ZSMALLOC \
 	CONFIG_ZRAM \
 	CONFIG_ZRAM_DEBUG=n \
 	CONFIG_ZRAM_WRITEBACK=n \
-	CONFIG_ZSMALLOC_STAT=n
+	CONFIG_ZSMALLOC_STAT=n \
+	CONFIG_ZRAM_BACKEND_LZ4=y \
+	CONFIG_ZRAM_BACKEND_LZO=y \
+	CONFIG_ZRAM_BACKEND_ZSTD=y
   FILES:= \
 	$(LINUX_DIR)/mm/zsmalloc.ko \
 	$(LINUX_DIR)/drivers/block/zram/zram.ko
